design/userdefined-rootdefinehints.md

```diff
@@ -0,0 +1,101 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# userdefined-rootdevicehints
+
+## Status
+
+provisional
+
+## Table of Contents
+
+<!--ts-->
+   * [userdefined-rootdevicehints](#userdefined-rootdevicehints)
+      * [Status](#status)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+         * [Non-Goals](#non-goals)
+      * [Proposal](#proposal)
+         * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Work Items](#work-items)
+         * [Dependencies](#dependencies)
+         * [Test Plan](#test-plan)
+         * [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+         * [Version Skew Strategy](#version-skew-strategy)
+      * [References](#references)
+
+<!-- Added by: dhellmann, at: Tue Jun 25 14:54:44 EDT 2019 -->
+
+<!--te-->
+
+## Summary
+
+This document explains how rootdevicehints could be given as user defined parameters as part BaremetalHostSpec. The idea is that user would have a possibilty to defines selective constrains for root device selection.
+
+## Motivation
+
+We need to be able select root device, when there is multiple devices available so that device is not chosen randomly
+
+### Goals
+
+1. Decide root device hints are needed be supported.
+2. Agree on the way haw information is propagated to ironic through BMO
+
+
+### Non-Goals
+
+The complete set of root device hints as input is not needed only those seen important.
+
+## Proposal
+
+### Implementation Details/Notes/Constraints
+
+- BaremetalHostSpec need to be updated to support list of following root device hints:
+* vendor
+* size
+* wwn
+
+The above parameters are optional in BaremetalHostSpec. If user is using legacy Hardware profiles, the root device hints hardcoded to code can be overwritten or appended with proposed root device hints
```

Comment:
We should go ahead and start the process of deprecating the profiles feature. That means exposing the fields used by the profiles, like HCTL and deviceName.

Reply:
I added devicename and HCTL into a list. Could deprication be a separate PR?

Reply:
Yes, I just meant that we should say in this document that this is a step toward deprecating the profile feature. My question about precedence applies to the profile, too. We should say that a user-provided hint overrides the hardware profile setting.

Reply:
I now updated a document and mentioned that